### DISCUSSION TITLE
The component doesn't work after being compiled in Nuxt?

### DISCUSSION THREAD

**Discussion by tanjie-ola** [nuxt/nuxt#28278] [Questions] [answered]
The repository address where the issue can be reproduced: [https://github.com/tanjie-ola/nuxt-render](https://github.com/tanjie-ola/nuxt-render)

"I just encountered an issue and I'm not sure if it's related to Nuxt. Can anyone help me with this? Thanks.

1. Download 
   ```bash
   git clone https://github.com/tanjie-ola/nuxt-render.git
   ```
2 dev is OK
```bash
   pnpm run dev
```
3. build 

```bash
   pnpm run build && pnpm run preview
```

In the Vue project, both development and production work fine. In the Nuxt project, development works fine, but after building, it doesn't work. The rendered content is not being compiled correctly. The code includes a table based on Arco components and a custom component I wrote, neither of which are functioning properly.

dev:
![image](https://github.com/user-attachments/assets/a1404075-2257-4ef5-9b2f-d57b13053ec3)

build:
![image](https://github.com/user-attachments/assets/21bb6f01-adc3-47ae-80b3-387e41c7be57)

**Comment by Davvii1** [comment]
Hi! I recreated the steps same as you described in your question and the output is a functional table: 
![image](https://github.com/user-attachments/assets/2125ef4e-ab15-4aeb-82eb-18ac2c9de97c)
The only thing I did differently was to modify `nuxt.config.ts`, adding the modules section like this: 
```ts
export default defineNuxtConfig({
  compatibilityDate: '2024-04-03',
  devtools: { enabled: true },
  modules: [
    'arco-design-nuxt-module'
  ]
})
```
check if you have this added to your nuxt config file and if this is already set up, let me know if you get an error output or something from console inside browser or terminal

**Reply by tanjie-ola** [reply]
Thank you for your response. I have already added the configuration, and it is the same as yours. Are you sure it's `pnpm run build && pnpm run preview`? 
I deployed the code to Vercel and encountered the same issue, so it shouldn't be a problem with my local environment.


![image](https://github.com/user-attachments/assets/2d63f35c-e5ac-4d32-9cbe-e7b1a1f748e7)

**Comment by xjccc** [comment, marked as answer]
I think.
```
import {Tag as ATag} from '@arco-design/web-vue'
```
And change your render function 

```
 return h(ATag, record.name)
```

**Reply by tanjie-ola** [reply]
Thank you for your reply. I tried it and it indeed works, but I have to write vNode. I still can't understand why JSX doesn't work. Is additional configuration needed?

**Reply by xjccc** [reply]
I don't know that .. I never use this case.. I think it's something wrong in config render for jsx..
